File: simple/cppvulnweb/code_exec/sub_class_test.cpp
#include "common/vuln_child.h"
#include <iostream>
#include <string>

// 代码执行漏洞 - 纯子类继承调用测试文件
// 测试场景：通过子类 VulnChild 继承父类 VulnParent 的代码执行漏洞方法

void code_exec_sub_class_test(const std::string& code) {
    std::cout << "=== 代码执行漏洞 - 子类继承调用测试 ===" << std::endl;

    VulnChild child;

    // 场景1：外部可控数据源（通过请求参数传入）
    // 调用链：VulnChild -> VulnParent::executeCode -> popen python3 -c（漏洞点）
    std::string result1 = child.executeCode(code);
    std::cout << "[外部参数] 代码执行结果: " << result1 << std::endl;

    // 场景2：硬编码无外部数据源
    std::string hardcoded_code = "print('hardcoded_code_exec_test')";
    std::string result2 = child.executeCode(hardcoded_code);
    std::cout << "[硬编码] 代码执行结果: " << result2 << std::endl;
}

File: simple/javavulnweb/src/main/java/com/vulnweb/code_exec/SubClassTest.java
package com.vulnweb.code_exec;

import com.vulnweb.common.VulnChild;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * 代码执行漏洞 - 纯子类继承调用测试文件
 * 通过 VulnChild 子类调用父类代码执行漏洞方法
 */
@RestController
public class SubClassTest {
    private VulnChild child = new VulnChild();

    /**
     * 测试代码执行：硬编码脚本
     */
    @GetMapping("/code/sub/hardcoded")
    public String testHardcoded() {
        String result = child.executeCode("1 + 2");
        return "子类继承硬编码代码执行结果: " + result;
    }

    /**
     * 测试代码执行：外部参数
     */
    @GetMapping("/code/sub/param")
    public String testParam(@RequestParam String script) {
        // 调用链：请求参数 -> SubClassTest -> VulnChild -> VulnParent.codeExec
        String result = child.executeCode(script);
        return "子类继承带参代码执行结果: " + result;
    }
}